RtspServer/Classes/SourceMedia.cs

```diff
@@ -239,15 +238,15 @@ public SourceMedia(string name, Uri source, NetworkCredential sourceCredential,
 
         internal void OnFrameDecoded(System.Drawing.Image decoded)
         {
-            if (DecodeFrames && decoded != null && FrameDecoded != null)
+            if (DecodeFrames && decoded != null)
```

Comment:
Why not leave the check for the delegate in this pattern and remove the conditional check e.g.

```
 if (DecodeFrames && decoded != null && DataDecoded != null)                 DataDecoded.Invoke(this, decoded);
```